sycl/include/sycl/ext/oneapi/experimental/enqueue_functions.hpp

```diff
@@ -357,6 +288,75 @@ void nd_launch(queue Q, launch_config<nd_range<Dimensions>, Properties> Config,
   });
 }
 
+// Free function kernel enqueue functions
+template <auto *Func, typename... ArgsT>
+void single_task(queue Q, kernel_function_s<Func> KernelFunc, ArgsT &&...Args) {
+  (void)KernelFunc;
+  submit(Q, [&](handler &CGH) {
+    single_task(CGH, KernelFunc, std::forward<ArgsT>(Args)...);
+  });
```

Comment:
Do we have a `submit_direct*` version of this?

Reply:
We can have it in `detail::` still. Also, `queue::*` itself can act as `submit_direct`.

Reply:
I chose to keep it like this rather than use something like `queue::single_task` in order to not deviate from the implementation of the other functions in this file where the queue version of the function delegates to the handler version of the function.

Reply:
> in order to not deviate from the implementation of the other functions in this file

How are we going to ever improve with such reasoning?

Reply:
Correct. But it means that we need to create this infrastructure. My initial understanding is that the implementation of the free function kernel enqueue should be implemented using handler-less path (`submit_kernel_direct_*`). Do we have examples where we really need a handler in this case?

Reply:
Tagging @gmlueck. If need be, I will revamp this PR to implement handler-less path for free function kernels but I'd like to consult the spec writers first.